source/MaterialXFormat/Util.cpp

```diff
@@ -148,6 +148,10 @@ void flattenFilenames(DocumentPtr doc, const FileSearchPath& searchPath, StringR
         }
 
         FilePath unresolvedValue(valueElem->getValueString());
+        if (unresolvedValue.asString().empty())
```

Comment:
Shouldn't you use isEmpty() here instead of converting?

Reply:
Why not just do:
if (!valueElem || valueElem->getValueString().isEmpty() || valueElem->getType() != FILENAME_TYPE_STRING)
{
   continue;
}